woocommerce/woocommerce-ios
# Edit Product/Shipping/Price/Inventory Settings: handled back navigation while there are outstanding changes

Fixes #1640 

## Description
If there are outstanding changes on the Product Title/Description/Shipping/Price/Inventory Settings, now we show this action sheet when the user navigates back from the settings screen in case they intend to save the changes.

## Changes
- New `UIAlertController` helpers extension useful to present the Save Changes ActionSheet
- New extension and protocol in `UINavigationController` extension to handle `UINavigationBar`'s **Back** button action
- Managed Save Changes Action Sheet in `AztecEditor`
- Managed Save Changes Action Sheet in `TextViewViewController`
- Managed Save Changes Action Sheet in all the controllers for editing product settings 

## Testing
1) Open a product detail
2) For every editable field (Title, Description, Price, Shipping, Inventory) try to edit something, and press the back button.
3) Try to discard change, and make sure that the new value was not shown in the product detail.
4) Repeat point 2, and now try to save changes. Make sure that every value was correctly saved and showed in the product detail.

## Screenshots
| Title             |  Description |  Price |
:-------------------------:|:-------------------------:|:-------------------------:
![Simulator Screen Shot - iPhone 11 Pro - 2020-01-14 at 15 10 10](https://user-images.githubusercontent.com/495617/72352910-69e03980-36e3-11ea-9637-fd25f969476c.png) | ![Simulator Screen Shot - iPhone 11 Pro - 2020-01-14 at 15 10 28](https://user-images.githubusercontent.com/495617/72352930-75336500-36e3-11ea-93bf-c2c1915be104.png) | ![Simulator Screen Shot - iPhone 11 Pro - 2020-01-14 at 15 10 36](https://user-images.githubusercontent.com/495617/72352954-7f556380-36e3-11ea-812c-f5d7ca9d2062.png)

| Shipping             |  Inventory |
:-------------------------:|:-------------------------:
![Simulator Screen Shot - iPhone 11 Pro - 2020-01-14 at 15 10 52](https://user-images.githubusercontent.com/495617/72353263-0a365e00-36e4-11ea-8f44-3326b8e85203.png) | ![Simulator Screen Shot - iPhone 11 Pro - 2020-01-14 at 15 11 01](https://user-images.githubusercontent.com/495617/72353273-0e627b80-36e4-11ea-8cd5-383603fe5676.png)

Update release notes:
